consensus/core/src/block.rs

```diff
@@ -154,4 +146,34 @@ impl fmt::Debug for BlockDigest {
     }
 }
 
+/// Signature of block digest by its author.
+#[allow(unused)]
+pub(crate) type BlockSignature = NetworkKeySignature;
+
+/// Unverified block only allows limited access to its content.
+#[allow(unused)]
+#[derive(Deserialize)]
+pub(crate) struct SignedBlock {
```

Comment:
Do we try to use naming around actions? Ex Signed, Verified etc. Otherwise `UnverifiedBlock` might also make sense since a priori all the blocks are signed - the signature validity is the one in question.

Reply:
I'm using the convention from Sui i.e. SignedTransaction, VerifiedTransaction etc. We can iterate on this later.